assignment1/layers.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+
+def linear_forward(A, W, b):
+    """
+    Description: Implement the linear part of a layer's forward propagation.
+    
+    input:
+        A – the activations of the previous layer
+        W – the weight matrix of the current layer (of shape [size of current layer, size of previous layer])
+        B – the bias vector of the current layer (of shape [size of current layer, 1])
+
+    Output:
+        Z – the linear component of the activation function (i.e., the value before applying the non-linear function)
+        linear_cache – a dictionary containing A, W, b (stored for making the backpropagation easier to compute)
+    """ 
+    n_activations = A.shape[0]
+    Z = np.dot(A.reshape(n_activations,-1), W) + b 
```

Comment:
why do you use this reshape? I think A should be a matrix - for a batch of samples no?

Reply:
Okay, so what is A.shape? 
I got my inspiration from my cs231n code:
https://github.com/elisim/cs231n/blob/7e0e8ec8cb675b1a52398bcb59576d995e8f8ce8/assignment2/cs231n/layers.py#L5